### PR TITLE
Fix dashboard crash when GitHub returns no contribution data

### PR DESCRIPTION
**Pull request by aniknafs**
## Summary

The dashboard's contribution calendar can take down the whole app.

`react-activity-calendar` validates its data **during render** and throws
`Activity data must not be empty.` whenever the contributions API answers
`200` with an empty list:

```
$ curl 'https://github-contributions-api.jogruber.de/v4/<user>?y=2026'
{"total":{},"contributions":[]}
```

That happens for any account with no public contributions in the requested
year — EMU/enterprise accounts, brand new accounts, or anyone whose work
lives on a GitHub Enterprise instance. Because the throw happens in render,
it escapes to the app-level `ErrorBoundary`, so a *decorative widget*
replaces the entire dashboard with the crash screen a few seconds after
launch, once the calendar data loads.

Observed on a fresh `pnpm tauri dev` run:

```
ERROR ship_studio_lib::logging: Activity data must not be empty. source="frontend"
  componentStack: ActivityCalendar → GitHubCalendar → ProjectList → ProjectsView → App → ErrorBoundary
```

**Fix:** wrap the third-party calendar in its own error boundary. Any throw
now hides just the calendar, logs a warning, and leaves the dashboard
intact. Same log line after the fix — note `WARN`, and the app stays usable:

```
WARN ship_studio_lib::logging: GitHub contribution calendar failed to render, hiding it
  {"error":"Activity data must not be empty.","username":"<user>"}
```

This also backstops the `color-mix` failure already documented in this file,
should that theme workaround ever regress — the widget degrades instead of
white-screening the app.

One subtlety worth calling out for review: the failure is tracked as
`failedFor: string | null` (the username it blew up for) rather than a
boolean. The existing "reset on username change" effect runs *after* the
boundary's `componentDidCatch` on mount, so a boolean cleared there would
immediately un-hide a calendar that had just failed, and it would re-throw
in a loop.

## Test plan

- [x] Added `GitHubCalendar.test.tsx`: the throwing case asserts the widget
      unmounts **and** that sibling dashboard content survives (i.e. the error
      never reaches the app boundary); the healthy case asserts the calendar
      still renders.
- [x] Reproduced live in `pnpm tauri dev` with an account that has no public
      contributions — before: full crash screen; after: dashboard loads
      normally with the calendar absent and a single `WARN` in the log.
- [x] Verified the healthy path live with an account that *does* have
      contributions — calendar renders as before, no warning.

## CI gates

- [x] `pnpm check:all && pnpm test:run` passes locally
      (typecheck · lint:strict · format:check · check:patterns · check:loc;
      112 test files, 1530 passed / 4 skipped)
- [ ] `pnpm rust:test` — 2 pre-existing failures on my machine, unrelated to
      this PR (it changes **zero** Rust files; `git diff main...HEAD` touches
      only the two files above). For the record:
      `proxy::tests::connect_retry_exhausts_budget_with_refused_kind_for_dead_port`
      passes when run serially — it looks like a port-reuse race under
      parallel test execution — and
      `commands::git::tests::run_git_net_retrying_index_lock_passes_other_failures_through`
      fails its `< 500ms` timing assertion in this environment (a bare
      `git checkout` with the same credential-helper config measures ~30ms
      here, so the budget is being spent elsewhere in the helper's setup).
      Happy to open a separate issue if these aren't already known.
- [x] Also ran the two CI gates that aren't part of that command:
      `bash scripts/check-config-integrity.sh` (the Config Integrity job) and
      the coverage floor step — 31.6% lines / 81.3% branches, comfortably
      above the 4% / 15% thresholds CI enforces.

<details>
<summary><strong>Patterns checklist</strong></summary>

**Frontend**
- ~~New modals use `<ModalFrame>`~~ — n/a
- ~~New buttons use `<Button variant=…>`~~ — n/a
- ~~New async state uses `useAsyncState` / `useInvoke`~~ — n/a, no new async
  state; the added state is a render-failure latch
- ~~New polling uses `usePolling`~~ — n/a
- ~~Modal state uses `useModal('id')`~~ — n/a

An error boundary requires a class component (`getDerivedStateFromError` /
`componentDidCatch` have no hook equivalent), so this one is local to the
file. It's deliberately *not* the shared `ErrorBoundary` — that one is the
app-level crash screen, whereas this needs to render `null` and let the rest
of the dashboard carry on.

**CSS**
- [x] No CSS changed

**Rust**
- [x] No Rust changed

**Tests**
- [x] Added tests for non-trivial logic

</details>

## Summary by CodeRabbit

* **Bug Fixes**
  * The GitHub contribution calendar now fails gracefully if it encounters a rendering error.
  * When the calendar cannot load, it is hidden while the rest of the dashboard remains available.
  * The calendar resets correctly when viewing a different username.

### CODE REVIEW

File: src/components/dashboard/GitHubCalendar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  GitHubCalendar: ({
+    transformData,
+  }: {
+    transformData: (data: Array<{ date: string; count: number; level: 0 }>) => unknown;
+  }) => {
+    transformData([]);
+    if (libState.shouldThrow) {
+      throw new Error('Activity data must not be empty.');
+    }
+    return <div data-testid="calendar-graph" />;
```
[the remark this answers is not in the document]
Comment (aniknafs) [reply src/components/dashboard/GitHubCalendar.test.tsx]: Good catch — fixed in e1600f5.

The healthy case now gets a real contribution record; the failing case keeps the empty list, since that empty list is exactly what the API returns for accounts with no public contributions and is what makes the library throw.

Adding the valid data surfaced a second problem with the mock, so I went a bit further than the suggestion: it was calling `transformData` synchronously during the first render, whereas the real library renders nothing while fetching and only calls it in a later pass. That timing is load-bearing here — the parent resets `dataLoaded` in its own mount effect, so a synchronously-delivered payload gets stomped by the reset and the calendar never leaves its skeleton. The mock now delivers data from an effect, and the healthy test asserts the widget is actually *visible* rather than merely mounted, which is what caught it.

Also sanity-checked that the failure test earns its keep: with the error boundary removed, it fails; with it restored, it passes.
